web/src/api/getState.ts

```diff
@@ -1,10 +1,60 @@
 import type { UseQueryOptions, UseQueryResult } from '@tanstack/react-query';
 import { useQuery } from '@tanstack/react-query';
-import type { GridState, TimeAverages } from 'types';
+import generateTopos from 'features/map/map-utils/generateTopos';
+import type { GridState, MapGrid, MapZone, TimeAverages } from 'types';
 import { getBasePath, getHeaders, QUERY_KEYS, REFETCH_INTERVAL_MS } from './helpers';
+import { featureCollection } from '@turf/turf';
 
-const getState = async (timeAverage: string): Promise<GridState> => {
-  const path = `/v5/state/${timeAverage}`;
+export function getCO2IntensityByMode(
+  zoneData: { co2intensity: number; co2intensityProduction: number },
+  electricityMixMode: string
+) {
+  return electricityMixMode === 'consumption' ? zoneData.co2intensity : zoneData.co2intensityProduction;
+}
+
+const getSelectedDatetime = (dateTimes: Array<string>) => {
+  return dateTimes[dateTimes.length - 1];
+};
+const getLength = (coordinate?: [number, number]) => (coordinate ? coordinate.length : 0);
+
+const mapZonesToGrid = ({ data, callerLocation }: GridState, getCo2colorScale: (co2intensity: number) => string) => {
+  const keys = Object.keys(data.zones) as Array<keyof GridState>;
+  const geographies = generateTopos();
+  const selectedDateTime = getSelectedDatetime(data.datetimes);
+  if (!keys) {
+    return [] as MapZone[];
+  }
+  const zones = keys
+    .map((key) => {
+      if (!geographies[key]) {
+        return;
+      }
+      const zoneData = data.zones[key][selectedDateTime];
+      const co2intensity = zoneData ? getCO2IntensityByMode(zoneData, 'consumption') : undefined; //TODO get mode
+      const fillColor = co2intensity ? getCo2colorScale(co2intensity) : undefined;
+      return {
+        type: 'Feature',
+        geometry: {
+          ...geographies[key].geometry,
+          coordinates: geographies[key].geometry.coordinates.filter((element: [number, number] | undefined) =>
+            getLength(element)
+          ),
```

Comment:
This shouldn't actually be required AFAIK, as we're doing this in `generate-geo-file.js`:
```
  if (longitudes.length === 0 || latitudes.length === 0) {
    console.error(`ERROR: Found ${longitudes.length} longitudes and ${latitudes} latitudes for zoneName ${zoneName}`);
    process.exit(1);
  }
```

But in order to not spend too much time on it now, we can also just use this suggestion:
```suggestion
          // TODO: Figure out if we still need this check, as invalid/empty coordinates should be removed during buildtime generation
          coordinates: geographies[key].geometry.coordinates.filter((element: [number, number] | undefined) =>
            getLength(element)
          ),
```